### PR TITLE
Add kabinet kasir

### PR DESCRIPTION
**Pull request by conanvster**
Без блокировки матча отмеченного как абонемент.
Начал путаться в коде, сделаю отдельным коммитом

### CODE REVIEW

File: client/app/admin/season-ticket/season-ticket.controller.js [hunk 1]
```diff
@@ -20,6 +22,11 @@ export default class SeasonTicketController {
     this.errorMessageSeat = '';
```
Comment (DmytroLypovetskiy) [inline client/app/admin/season-ticket/season-ticket.controller.js]: second this.errorMessageSeat?

File: client/app/admin/season-ticket/season-ticket.controller.js [hunk 2]
```diff
@@ -20,6 +22,11 @@ export default class SeasonTicketController {
...
   }
 
+  loadPreSaleAbonementTickets() {
+    this.ticketsService.fetchReservedSeats().then(response => this.preSaleAbonementTickets = response.data);
+    this.errorMessageSeat = '';
```
Comment (DmytroLypovetskiy) [inline client/app/admin/season-ticket/season-ticket.controller.js]: 3rd this.errorMessageSeat?

File: client/app/cashbox/abonement-ticket/abonement-ticket.component.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import template from './abonement-ticket.html';
+import AbonementTicketController from './abonement-ticket.controller';
+let abonementTicketComponent = {
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/abonement-ticket/abonement-ticket.component.js]: Why is not `const`?

File: client/app/cashbox/abonement-ticket/abonement-ticket.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      </td>
+      <td>
+        {{ticket.row}}
+      </td>
+      <td>
+        {{ticket.seat}}
+      </td>
+      <td>
+        {{ticket.reservedUntil | date}}
+      </td>
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/abonement-ticket/abonement-ticket.html]: Maybe it's better to use `ng-repeat` here?

File: client/app/cashbox/cashbox.controller.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+export default class CashboxController {
+    constructor(SeasonTicketService, CashboxService) {
+        'ngInject';
+        this.seasonTicketService = SeasonTicketService;
+        this.cashboxService = CashboxService;
+        this.errorMessageSeat;
+        this.preSellTicket = null;
+        this.ticket = {};
+        this.date = new Date(2018, 6, 1);
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/cashbox.controller.js]: What happens this date? 2018, 6, 1

File: client/app/cashbox/cashbox.controller.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            .then(response => {
+        this.preSellTicket = response;
+});
+
+}
+
+createSeasonTicket() {
+    console.log('3475657817460674', this.preSellTicket);
+    let slug = 's' + this.preSellTicket.seat.sector + 'r' + this.preSellTicket.seat.row + 'st' + this.preSellTicket.seat.seat;
+    let ticket =  this.preSellTicket.seat;
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/cashbox.controller.js]: `seat.seat` doesn't' look great.

File: client/app/cashbox/cashbox.controller.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    ticket.reservedUntil = this.date;
+    ticket.accessCode = this.preSellTicket.accessCode;
+    this.seasonTicketService.createSeasonTicket(ticket, slug)
+        //.then(() => {   this.loadSeasonTickets(); })
+        .then(() => {
+                    this.cashboxService.setTicketUsed(this.preSellTicket._id)
+                    })
+        .catch((err) => {
+            if (err.status === 409) {
+            this.errorMessageSeat = 'Это место уже занято.';
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/cashbox.controller.js]: use one line when having simple `if` statement

 if (err.status === 409) this.errorMessageSeat = 'Это место уже занято.';

File: client/app/cashbox/cashbox.html [hunk 8]
```diff
@@ -0,0 +1,43 @@
+<div class="container">
+
+  <div class="col-md-4">
+    <label>Регистрировать билет</label>
+    <form class="form-inline">
+      <div class="form-group">
+        <input type="text" class="form-control" placeholder="Штрихкод билета" ng-model="cashbox.accessCode">
+      </div>
+      <button type="submit" class="btn btn-default" ng-click="cashbox.regTicket(cashbox.accessCode)">Найти</button>
+    </form>
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/cashbox.html]: why don't u use `ng-submit` directive for form?

File: client/app/cashbox/cashbox.html [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    <hr>
+
+    <form ng-show="cashbox.preSellTicket">
+      <div  class="form-group">
+        <label>Место</label><br>
+        Трибуна  {{cashbox.preSellTicket.seat.tribune}}
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/cashbox.html]: just wrap this text into `<p>` tag instead of using `<br>`

File: client/app/cashbox/cashbox.module.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import CashboxController from './cashbox.controller';
+import AbonementTicketComponent from './abonement-ticket/abonement-ticket.component';
+import CashboxService from './cashbox.service';
+
+let cashboxModule = angular.module('metalistTicketsApp.cashbox', [])
```
Comment (DmytroLypovetskiy) [inline client/app/cashbox/cashbox.module.js]: it's better to use `const`. `let` is for updating variables

File: server/api/seasonTicket/index.js [hunk 11]
```diff
@@ -6,7 +6,7 @@ import * as auth from '../../auth/auth.service.js';
 
 let router = express.Router();
```
Comment (DmytroLypovetskiy) [inline server/api/seasonTicket/index.js]: `const`

File: server/api/ticket/ticket.service.js [hunk 12]
```diff
@@ -45,6 +45,11 @@ export function getByTicketNumber(ticketNumber) {
   return Ticket.findOne({ticketNumber: ticketNumber});
 }
 
+export function getByAccessCode(accessCode) {
+  console.log('accessCode', accessCode);
+  return Ticket.findOne({accessCode: accessCode});
+}
+
 export function randomNumericString(length) {
   let chars = '0123456789';
```
Comment (DmytroLypovetskiy) [inline server/api/ticket/ticket.service.js]: why? :)

File: server/api/seat/seat.controller.js [hunk 13]
```diff
@@ -13,6 +13,7 @@ export function getReservedSeats(req, res) {
   return Promise.all([
     seatService.getReservedSeats(matchId, sector),
     seasonTicketService.getActiveBlocksBySector(sector)
+    seasonTicketService.getActiveBlocksBySector(sector)
```
Comment (Skadabr) [inline server/api/seat/seat.controller.js]: where is comma?

Comment (Skadabr) [inline server/api/seat/seat.controller.js]: it should be removed at all I hope as duplicate

File: server/api/ticket/ticket.service.js (hunk 12 above)
Comment (conanvster) [inline server/api/ticket/ticket.service.js]: I do not know how it works, I'll fix it.

File: client/app/cashbox/abonement-ticket/abonement-ticket.html (hunk 4 above)
Comment (conanvster) [inline client/app/cashbox/abonement-ticket/abonement-ticket.html]: There is used ng-repeat

File: client/app/admin/season-ticket/season-ticket.controller.js (hunk 2 above)
Comment (conanvster) [inline client/app/admin/season-ticket/season-ticket.controller.js]: Clearing a Variable

File: client/app/admin/admin-users/admin-users.html [hunk 14]
```diff
@@ -14,6 +14,7 @@
     <span class="text-muted">{{user.email}}</span>
     <span class="btn-group-sm pull-right" ng-hide="user.role=='admin'">
         <a class="role" ng-click="$ctrl.setRole(user, 'steward')"><span class="text-warning">Steward</span></a> |
+        <a class="role" ng-click="$ctrl.setRole(user, 'cashier')"><span class="text-warning">Cashier</span></a> |
         <a class="role" ng-click="$ctrl.setRole(user, 'moderator')"><span class="text-success">Moderator</span></a> |
         <a class="role" ng-click="$ctrl.setRole(user, 'user')"><span class="text-info">User</span></a>
```
Comment (conanvster) [inline client/app/admin/admin-users/admin-users.html]: Refactoring is planned

File: client/app/cashbox/cashbox.html (hunk 8 above)
Comment (conanvster) [inline client/app/cashbox/cashbox.html]: I'm learning to use

File: client/app/admin/admin-users/admin-users.html (hunk 14 above)
Comment (Skadabr) [inline client/app/admin/admin-users/admin-users.html]: form mobile this list will be not useful. keeping as is

File: client/app/admin/season-ticket/season-ticket.controller.js (hunk 1 above)
Comment (Skadabr) [inline client/app/admin/season-ticket/season-ticket.controller.js]: maybe not optimal logic but its resetting error message. will add to tech debt

File: client/app/cashbox/cashbox.html (hunk 8 above)
Comment (Skadabr) [inline client/app/cashbox/cashbox.html]: totally agree
ngSubmit ensures that the form doesn’t submit when the handler code throws (which is the default behaviour of submit) and causes an actual http post request. will use ngSubmit instead as this is the best practice:
created story in trello for this refactoring
https://trello.com/c/v0Mj7cnl/139-refactor-client-app-cashbox-cashboxhtml-to-use-ng-submit

File: client/app/admin/season-ticket/season-ticket.controller.js (hunk 1 above)
Comment (DmytroLypovetskiy) [inline client/app/admin/season-ticket/season-ticket.controller.js]: so it's better to create a method that return a message from argument or reset it if argument is not exist;
```
this.updateEerrorMessage(msg) {
return (msg) ? msg : '';
}
```

than just do `this.errorMessageSeat = this.updateEerrorMessage() ;`
or `this.updateEerrorMessage(msg);`

File: client/app/admin/season-ticket/season-ticket.controller.js [hunk 15]
```diff
@@ -1,10 +1,12 @@
 export default class SeasonTicketController {
 
-  constructor(SeasonTicketService) {
+  constructor(SeasonTicketService, TicketsService) {
     'ngInject';
     this.seasonTicketService = SeasonTicketService;
+    this.ticketsService = TicketsService;
```
Comment (conanvster) [inline client/app/admin/season-ticket/season-ticket.controller.js]: -

File: client/app/cashbox/cashbox.html [hunk 16]
```diff
@@ -0,0 +1,40 @@
+<div class="container">
+
+  <div class="col-md-4">
+    <label>Регистрировать билет</label>
+    <form class="form-inline">
+      <div class="form-group">
+        <input type="text" class="form-control" placeholder="Штрихкод билета" ng-model="cashbox.accessCode">
+      </div>
+      <button type="submit" class="btn btn-default" ng-click="cashbox.regTicket(cashbox.accessCode)">Найти</button>
```
Comment (conanvster) [inline client/app/cashbox/cashbox.html]: $ctrl

**Pull request merged**